src/Http/Routing/src/Matching/DfaMatcher.cs

```diff
@@ -34,11 +34,6 @@ public sealed override Task MatchAsync(HttpContext httpContext, EndpointSelector
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            if (context == null)
-            {
-                throw new ArgumentNullException(nameof(context));
-            }
-
```

Comment:
Take a look here: https://github.com/aspnet/AspNetCore/pull/9774/files#diff-1473cfa07c9b9a3090e60b371891a1f4L310 This has to do something different now to process short-circuits.